src/request-link-encoding/RequestLinkEncoding.ts

```
@@ -374,21 +372,27 @@ function getEthereumBlochainName(chainId?: number) {
     if (!chainId) throw new Error(`Unsupported chain: ${chainId}`);
 
     switch (chainId) {
-        case ETHEREUM_CHAIN_ID.ETHEREUM_MAINNET:
-        case ETHEREUM_CHAIN_ID.ETHEREUM_GOERLI_TESTNET:
-            return 'ethereum';
         case ETHEREUM_CHAIN_ID.POLYGON_MAINNET:
         case ETHEREUM_CHAIN_ID.POLYGON_MUMBAI_TESTNET:
             return 'polygon';
+        case ETHEREUM_CHAIN_ID.ETHEREUM_MAINNET:
+        case ETHEREUM_CHAIN_ID.ETHEREUM_GOERLI_TESTNET:
         default:
-            return undefined;
+            return 'ethereum';
     }
 }
 
 function validEthereumAddress(address: string): boolean {
     return /^0x[a-fA-F0-9]{40}$/.test(address);
 }
 
-function getSupportedTokens(chainId: number) {
-    return SUPPORTED_TOKENS[chainId as keyof typeof SUPPORTED_TOKENS];
+function isNativeToken(currency: Currency): boolean {
+    return currency === Currency.ETH;
+}
+
+// Return the contract address for the given chainId and currency. If no contract address is known for the given
+// chainId and currency, return empty string.
+function getContractAddress(chainId: number, currency: Currency) {
+    const tokens = SUPPORTED_TOKENS[chainId as keyof typeof SUPPORTED_TOKENS];
+    return tokens[currency as unknown as keyof typeof tokens] as string || '';
```

I'm still of the opinion this should throw instead of returning an empty string. An empty string in the URL means an invalid URL I think. @danimoh wdyt?